tests/driver_apa102/main.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for the APA102 LED strip driver
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <string.h>
+#include <stdio.h>
+
+#include "xtimer.h"
+#include "color.h"
+
+#include "apa102.h"
+#include "apa102_params.h"
+
+/**
+ * @brief   Switch to the next LED every 10ms
+ */
+#define STEP        (10 * US_PER_MS)
+
+/**
+ * @brief   Interval for dimming colors
+ */
+#define DIM         (100 * US_PER_MS)
+
+/**
+ * @brief   Maximum intensity value
+ */
+#define BMAX        (31U)
```

Comment:
Actually, I used it badly here -> the driver expects an 8-bit alpha value and does the scaling internally (by cutting off the 3 LSB). So the correct behavior in the test app would be to step through the full alpha range [0:255] -> fixed.